cogs/piazza.py

```diff
@@ -270,8 +270,10 @@ def piazza_start(self):
             self.bot.d_handler.piazza_handler = PiazzaHandler(self.bot.piazza_dict["course_name"], self.bot.piazza_dict["piazza_id"], PIAZZA_EMAIL, PIAZZA_PASSWORD, self.bot.piazza_dict["guild_id"])
 
         # dict.get defaults to None so a key error is never raised
-        for ch in self.bot.piazza_dict.get("channels"):
-            self.bot.d_handler.piazza_handler.add_channel(int(ch))
+        channels = self.bot.piazza_dict.get("channels")
+        if channels:
+            for ch in channels:
+                self.bot.d_handler.piazza_handler.add_channel(int(ch))
```

Comment:
This seems like the same functionality as the original so I don't think is necessary

Reply:
If channels is None then the "for ch in channels" will raise an error.

Reply:
It looks like the issue is that if it couldn't get thee channels, then .get returns `None` and thus can't be iterated. If you wanted to make it slightly more concise, I think that you could do something like
```python3
for ch in self.bot.piazza_dict.get("channels", tuple()):
    self.bot.d_handler.piazza_handler.add_channel(int(ch))
```
since you can add a default result.

Reply:
the default retval seems to be a good solution